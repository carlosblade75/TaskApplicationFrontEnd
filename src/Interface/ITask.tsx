export interface ITask{
    description: string,
    id: number,
    isCompleted: boolean

}
