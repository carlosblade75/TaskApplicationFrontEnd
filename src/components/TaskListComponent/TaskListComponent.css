.title {
    margin-bottom: 15px;
    margin-top: 10px;
    color: rgb(24, 26, 158);
    text-decoration: underline;
}

.containerList {
    height: 300px;
    overflow: scroll;
    overflow-x: hidden;
    padding: 10px;
}

