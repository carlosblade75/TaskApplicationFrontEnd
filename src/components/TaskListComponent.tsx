import React, {useState, useEffect} from 'react';
import {ITask} from '../Interface/ITask'
import {TaskComponent} from '../components/TaskComponent';

type PropsTaskList = {
    title: string,
    taskList: ITask[]
}

export const TaskListComponent = (props: PropsTaskList) => {

    const {title, taskList } = props;

    return (

        <div>
          <p>{title} - ({taskList.length})</p>
  
          <ul>
            
            {taskList.length > 0 && taskList.map((task, index) => {
              return <TaskComponent key= {task.id} id = {task.id} description = {task.description} isCompleted = {task.isCompleted} />
            })}

          </ul>
  
        </div>
    )
}