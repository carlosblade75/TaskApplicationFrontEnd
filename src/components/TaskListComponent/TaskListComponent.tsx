import React from 'react';
import { ITask } from '../../Interface/ITask'
import { TaskComponent } from '../TaskComponent/TaskComponent';
import './TaskListComponent.css'
import PropTypes from "prop-types";

type PropsTaskList = {
    title: string,
    taskList: ITask[]
}

export const TaskListComponent = (props: PropsTaskList) => {

    const {title, taskList } = props;

    return (

      <>
          <h3 className="title">{title} - ({taskList.length})</h3>

          <div className="containerList">
  
            <ul className = "listTask">
              
              {taskList.length > 0 && taskList.map((task, index) => {
                return  (
                          <li key= {task.id} > 
                            <label>{index + 1}.</label>
                            <TaskComponent 
                                  id = {task.id} 
                                  description = {task.description} 
                                  isCompleted = {task.isCompleted} 
                            />
                          </li>
                        )
              })}

            </ul>

          </div>

      </>

    )
}

TaskListComponent.propTypes = {
  title : PropTypes.string.isRequired,
  taskList: PropTypes.arrayOf(PropTypes.shape({id: PropTypes.number, description: PropTypes.string, isCompleted: PropTypes.bool})).isRequired
}

TaskListComponent.defaultProps = {
  title: 'Task List',
  taskList:[]
}