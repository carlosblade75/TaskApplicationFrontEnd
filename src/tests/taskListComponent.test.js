import React from 'react'
import '@testing-library/jest-dom'
import { TaskListComponent } from "../components/TaskListComponent/TaskListComponent"
import { shallow } from 'enzyme'
import { ITask } from '../Interface/ITask'

describe('Testing for <TaskListComponent/>', () => {
    
    const wrapper = shallow( <TaskListComponent /> )

    test('should show <TaskListComponent/> properly', () => {
        
        expect(  wrapper ).toMatchSnapshot()
    })

    test('should show List empty and a default title', () => {

        const titleInComponent = wrapper.find('h3').text()
        const ulInComponent = wrapper.find('ul').text().trim()

        expect( titleInComponent ).toBe('Task List - (0)')
        expect( ulInComponent ).toBe('')
    })

    test('should show a list with a title and a task properly', () => {

        const tasks:ITask[] = [{id:1, description:'First task', isCompleted:false}]

        const title = `Pending tasks`

        const wrapper = shallow( <TaskListComponent taskList={ tasks } title={title} /> )

        const titleInComponent = wrapper.find('h3').text()

        expect (wrapper.find('li').exists() ).toBe(true)

        expect (wrapper.find('TaskComponent').length ).toBe(tasks.length)

        expect( titleInComponent ).toBe( `${title} - (${tasks.length})`)
    })
    
})
