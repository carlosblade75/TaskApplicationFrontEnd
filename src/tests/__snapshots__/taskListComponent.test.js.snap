// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for <TaskListComponent/> should show <TaskListComponent/> properly 1`] = `
<Fragment>
  <h3
    className="title"
  >
    Task List
     - (
    0
    )
  </h3>
  <div
    className="containerList"
  >
    <ul
      className="listTask"
    />
  </div>
</Fragment>
`;
